src/swtpm_setup/swtpm_setup.c

```diff
@@ -1561,12 +1563,20 @@ int main(int argc, char *argv[])
           curr_grp ? curr_grp->gr_name : "<unknown>",
           tmpbuffer);
 
+    certsdir = g_dir_make_tmp("swtpm_setup.certs.XXXXXX", &err);
```

Comment:
I think we should be able to restrict the creation of the temp dir with
`if (flags & (SETUP_EK_CERT_F | SETUP_PLATFORM_CERT_F) != 0) {...`

Reply:
Thanks. Let me use `if (flags & (SETUP_EK_CERT_F | SETUP_PLATFORM_CERT_F)) {` because `tpm2_create_ek_and_cert()` is using it.